lib/core/presentation/pages/scan_qr_code/sub_pages/claim_rewards_page/claim_rewards_page.dart

```diff
@@ -0,0 +1,101 @@
+import 'package:app/core/application/event/get_event_detail_bloc/get_event_detail_bloc.dart';
+import 'package:app/core/application/event/claim_rewards_bloc/claim_rewards_bloc.dart';
+import 'package:app/core/presentation/pages/scan_qr_code/sub_pages/claim_rewards_page/widgets/claim_rewards_listing.dart';
+import 'package:app/core/presentation/widgets/common/appbar/lemon_appbar_widget.dart';
+import 'package:app/core/presentation/widgets/loading_widget.dart';
+import 'package:app/gen/fonts.gen.dart';
+import 'package:app/i18n/i18n.g.dart';
+import 'package:app/theme/spacing.dart';
+import 'package:app/theme/typo.dart';
+import 'package:auto_route/auto_route.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+@RoutePage()
+class ClaimRewardsPage extends StatelessWidget {
+  final String userId;
+  const ClaimRewardsPage({
+    super.key,
+    @PathParam("userId") required this.userId,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final t = Translations.of(context);
+    final colorScheme = Theme.of(context).colorScheme;
+    String eventId = context.watch<GetEventDetailBloc>().state.maybeWhen(
+              fetched: (event) => event.id,
+              orElse: () => "",
+            ) ??
+        "";
+    return MultiBlocProvider(
+      providers: [
+        BlocProvider(
```

Comment:
can we move up BlocProvider 1 widget tree level ? 
ClaimRewardPage + BlocProvider => ClaimRewardPageView